macros/src/lib.rs

```diff
@@ -2,8 +2,234 @@ extern crate proc_macro;
+            match &variant.fields {
+                Fields::Named(FieldsNamed { named, .. }) => {
+                    let field_idents = named
+                        .iter()
+                        .map(|field| field.ident.as_ref().unwrap())
+                        .collect::<Vec<_>>();
+
+                    let fields_count = named.iter().count();
+                    let enum_field_count = quote! {
+                        Self::#variant_name { #(#field_idents,)* } => {
+                            #fields_count
+                        }
+                    };
+
+                    let translate_fields =
+                        field_idents.iter().map(|f| quote!(#f: #f.translate(by)));
+                    let enum_translate = quote! {
+                        Self::#variant_name { #(#field_idents,)* } => {
+                            Self::#variant_name {
+                                #(#translate_fields,)*
+                            }
+                        }
+                    };
+
+                    let fields_index = field_idents
+                        .iter()
+                        .enumerate()
+                        .map(|(i, f)| quote!(#i => #f,));
+                    let enum_index = quote! {
+                        Self::#variant_name { #(#field_idents,)* } => {
+                            match index {
+                                #(#fields_index)*
+                                _ => panic!("Invalid index!"),
+                            }
+                        }
+                    };
+
+                    let enum_mut_index = enum_index.clone();
+
+                    let fields_draw = field_idents.iter().map(|f| quote!(#f.draw(display)?;));
+                    let enum_draw = quote! {
+                        Self::#variant_name { #(#field_idents,)* } => {
+                            #(#fields_draw)*
+                        }
+                    };
+
+                    (
+                        enum_field_count,
+                        enum_translate,
+                        enum_index,
+                        enum_mut_index,
+                        enum_draw,
+                    )
+                }
+                Fields::Unnamed(FieldsUnnamed { unnamed, .. }) => {
+                    let field_idents = unnamed
+                        .iter()
+                        .enumerate()
+                        .map(|(num, _)| format_ident!("__self_{}", num))
+                        .collect::<Vec<_>>();
+
+                    let fields_count = unnamed.iter().count();
+                    let enum_field_count = quote! {
+                        Self::#variant_name(#(#field_idents,)*) => {
+                            #fields_count
+                        }
+                    };
+
+                    let translate_fields = field_idents.iter().map(|f| quote!(#f.clone().translate(by), ));
```

Comment:
```suggestion
                    let translate_fields = field_idents.iter().map(|f| quote!(#f.translate(by), ));
```

This `clone` is not necessary, at least the example works without it. :)

Reply:
Move into a different PR and started a discussion on if `LinearLayout` should implement just `Clone` or `Clone + Copy`